packages/editor/src/components/provider/index.js

```diff
@@ -1,33 +1,37 @@
 /**
  * External dependencies
  */
-import { flow, map } from 'lodash';
+import { map } from 'lodash';
+import memize from 'memize';
 
 /**
  * WordPress Dependencies
  */
-import { createElement, Component } from '@wordpress/element';
-import { DropZoneProvider, SlotFillProvider } from '@wordpress/components';
-import { withDispatch } from '@wordpress/data';
+import { compose } from '@wordpress/compose';
+import { Component } from '@wordpress/element';
+import { withDispatch, withSelect } from '@wordpress/data';
 import { __ } from '@wordpress/i18n';
+import { BlockEditorProvider } from '@wordpress/block-editor';
 
 /**
  * Internal dependencies
  */
 import transformStyles from '../../editor-styles';
-
 class EditorProvider extends Component {
 	constructor( props ) {
 		super( ...arguments );
 
+		this.getBlockEditorSettings = memize( this.getBlockEditorSettings, {
```

Comment:
The memoization cache will be bust any time `edit-post`'s Editor component renders (currently if fixed toolbar, focus mode, or the saved post value changes) because we pass as an argument to `getBlockEditorSettings` the `settings` prop value produced by the parent `Editor` component. It's a new reference each time it is created:

https://github.com/WordPress/gutenberg/blob/3f1324b53cc8bb45d08d12d5321d6f88510bed09/packages/edit-post/src/editor.js#L28-L32

This feels very fragile, and the fact that the parent component is pure-by-`withSelect` and re-renders on specific options changes should not have an impact on how well this component is able to avoid generating its own settings to pass to `BlockEditorProvider`.

Reply:
It feels like the issue is in the parent component and it needs its own similar memoization?

Reply:
> It feels like the issue is in the parent component and it needs its own similar memoization?

I'm not positive what the best solution would be. It's similar to [issues we had with `InnerBlocks`'s `template` prop](https://github.com/WordPress/gutenberg/blob/c6255ed40efc3a8650723b6b2ce4a9f5485f6d26/packages/block-library/src/columns/index.js#L31-L40), and would impact anyone trying to render their own `EditorProvider`. In other cases I might have suggested trying to flatten the object to assign each as an individual prop. I'm not sure it makes much sense here.

For now, it may be enough to just add memoization to `edit-post`'s `Editor` as you suggest.

Reply:
Actually, this would have been a great useCase for `useMemo` :). I'd avoid for now as it's not the purpose of this PR, but would be good to update React now that there's at least an RC for react-native.